testtools/iothub_test/src/iothubtest.c

```diff
@@ -514,34 +514,49 @@ static char* CreateReceiveAddress(IOTHUB_VALIDATION_INFO* devhubValInfo, size_t
 static char* CreateReceiveHostName(IOTHUB_VALIDATION_INFO* devhubValInfo)
 {
     char* result;
-    size_t partner_host_len = strlen(devhubValInfo->partnerName) + strlen(devhubValInfo->partnerHost) + 2;
-    result = (char*)malloc(partner_host_len + 1);
-    if (result != NULL)
+    if (NULL == devhubValInfo)
     {
-        sprintf_s(result, partner_host_len + 1, "%s.%s", devhubValInfo->partnerName, devhubValInfo->partnerHost);
+        LogError("Invalid input parameter");
+        result = NULL;
     }
     else
     {
```

Comment:
Internal static function doesn't need to check the variable.

Reply:
Here and everywhere.

---
In reply to: [237168536](https://github.com/Azure/azure-iot-sdk-c/pull/734#discussion_r237168536) [](ancestors = 237168536)